internal/instrument/uninstrument.go

```diff
@@ -355,23 +357,46 @@ func removeGin(stmt dst.Stmt) bool {
 	if !ok {
 		return false
 	}
-	selexpr, ok := f.Fun.(*dst.SelectorExpr)
+	if !isUseMiddleware(f) {
+		return false
+	}
+	fun, ok := funcIdent(f.Args[0])
 	if !ok {
 		return false
 	}
-	if selexpr.Sel.Name != "Use" {
+	return fun.Name == "GinMiddleware" && fun.Path == "github.com/datadog/orchestrion/instrument"
+}
+
+// removeEcho returns whether a statement corresponds to orchestrion's Echo-middleware registration
+func removeEcho(stmt dst.Stmt) bool {
```

Comment:
Looks like `removeEcho` and `removeGin` are almost exactly the same function, only the middleware function name differs (`GinMiddleware` vs `EchoMiddleware`). I guess we can refactor it?
e.g
```golang
func removeGin(stmt dst.Stmt) bool {
    return removeMiddleware(stmt, "GinMiddleware")
}

func removeEcho(stmt dst.Stmt) bool {
    return removeMiddleware(stmt, "EchoMiddleware")
}
```

Reply:
@ahmed-mez I thought of that, but I'm not sure if all the frameworks we'll want to instrument follow this pattern `pkg.Use(instrument.PkgMiddleware)`. I'm going to do a quick check, to see if this `removeMiddleware` function won't lead to any confusion in the future, as it will always assume the mentioned pattern.

Reply:
Results for frameworks/routers listed in our Go Auto Instrumentation tracker.

They use `Use` for middlewares:

- gorilla/mux
- go-chi/chi
- gofiber/fiber.v2
- dimfeld/httptreemux.v5
- urfave/negroni
- gogearbox/gearbox
- zenazn/goji/web (I also saw this was superseded by goji/goji)

_Some of them has a `UseHandler` that maybe we should support too._

---

They don't use `Use` for middlewares:

- julienschmidt/httprouter: https://github.com/julienschmidt/httprouter#where-can-i-find-middleware-x

---

Conclusion:

I'm going to implement the refactor in this PR, naming the function as `removeUseMiddleware` to be consistent with other related functions. We'll see in another PR how to handle `UseHandler` function, as it's not present in Echo, and how to handle middlewares in HttpRouter.